compiler+runtime/test/cpp/jank/read/lex.cpp

```diff
@@ -1283,7 +1283,10 @@ namespace jank::read::lex
       {
         processor p{ R"(\ )" };
         native_vector<jtl::result<token, error_ref>> const tokens(p.begin(), p.end());
-        CHECK(tokens == make_results({ { make_error(kind::lex_incomplete_character, 0, 1) } }));
+        CHECK(tokens
+              == make_tokens({
+                { 0, 2, token_kind::character, "\\ "sv }
+        }));
```

Comment:
I think we need to test this on the parse side (in `test/cpp/jank/read/parse.cpp`), too. `\ ` and `\space` should provide the same character object. Right now, I'm not convinced they do.